source/solvers/heat_transfer_assemblers.cc

```diff
@@ -0,0 +1,607 @@
+#include <core/bdf.h>
+#include <core/time_integration_utilities.h>
+#include <core/utilities.h>
+
+#include <solvers/copy_data.h>
+#include <solvers/heat_transfer.h>
+#include <solvers/heat_transfer_assemblers.h>
+
+template <int dim>
+void
+HeatTransferAssemblerCore<dim>::assemble_matrix(
+  HeatTransferScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData &   copy_data)
+{
+  // Gather physical properties in case of mono fluids simulations (to be
+  // modified by cell in case of multiple fluids simulations)
+  double density              = physical_properties.density;
+  double specific_heat        = physical_properties.specific_heat;
+  double thermal_conductivity = physical_properties.thermal_conductivity;
+  double rho_cp               = density * specific_heat;
+  double alpha                = thermal_conductivity / rho_cp;
+
+  // Loop and quadrature informations
+  const auto &       JxW_vec    = scratch_data.JxW;
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const double       h          = scratch_data.cell_size;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  // Vector for the BDF coefficients
+  // The coefficients are stored in the following fashion :
+  // 0 - n+1
+  // 1 - n
+  // 2 - n-1
+  // 3 - n-2
+  std::vector<double> time_steps_vector =
+    simulation_control->get_time_steps_vector();
+
+  // Time steps and inverse time steps which is used for numerous calculations
+  const double dt  = time_steps_vector[0];
+  const double sdt = 1. / dt;
+
+  // Copy data elements
+  auto &strong_jacobian_vec = copy_data.strong_jacobian;
+  auto &local_matrix        = copy_data.local_matrix;
+
+  // assembling local matrix and right hand side
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    {
+      if (simulation_parameters.multiphysics.free_surface)
+        {
+          // Calculation of the equivalent physical properties at the
+          // quadrature point
+          density =
+            calculate_point_property(scratch_data.phase_values[q],
+                                     physical_properties.fluids[0].density,
+                                     physical_properties.fluids[1].density);
+
+          specific_heat = calculate_point_property(
+            scratch_data.phase_values[q],
+            physical_properties.fluids[0].specific_heat,
+            physical_properties.fluids[1].specific_heat);
+
+          thermal_conductivity = calculate_point_property(
+            scratch_data.phase_values[q],
+            physical_properties.fluids[0].thermal_conductivity,
+            physical_properties.fluids[1].thermal_conductivity);
+
+          // Useful definitions
+          rho_cp = density * specific_heat;
+          alpha  = thermal_conductivity / rho_cp;
+        }
```

Comment:
I guess this is why you passed the simulation parameters? I'm not sure this is the best approach. Maybe then pass the multiphysics parameters, but just pass the minimum needed to the class. I can understand that you want to keep these properties in a  if for now and that makes sense.